Analyzer/src/AnalyzeLepTrigger.cc

```diff
@@ -109,356 +97,158 @@ void AnalyzeLepTrigger::Loop(NTupleReader& tr, double weight, int maxevents, boo
         if(maxevents != -1 && tr.getEvtNum() >= maxevents) break;        
         if( tr.getEvtNum() % 1000 == 0 ) printf("  Event %i\n", tr.getEvtNum() ) ;
 
-
         //--------------------------------------------------
         //-- Print list of triggers (only if you want to see them)
         //--------------------------------------------------
 
-        //if( tr.getEvtNum() == 1 ) printTriggerList(TriggerNames); 
+        if( tr.getEvtNum() == 1 ) printTriggerList(TriggerNames); 
 
         // ------------------------
-        // -- Define weight
+        // -- Define eventweight
         // ------------------------
-        double weight               = 1.0;
         double eventweight          = 1.0;
         double leptonScaleFactor    = 1.0;
         double bTagScaleFactor      = 1.0;
         double htDerivedScaleFactor = 1.0;
         double topPtScaleFactor     = 1.0;
         double prefiringScaleFactor = 1.0;
         double puScaleFactor        = 1.0;
-        
+
         if(runtype == "MC")
         {
             if( !passMadHT ) continue; //Make sure not to double count DY events
             // Define Lumi weight
             const auto& Weight  = tr.getVar<double>("Weight");
             const auto& lumi = tr.getVar<double>("Lumi");
-            eventweight = lumi*Weight;
-            
+            const auto& puWeight = tr.getVar<double>("puWeightCorr");
+
             // Define lepton weight
             if(NGoodLeptons == 1)
             {
                 const auto& eleLepWeight = tr.getVar<double>("totGoodElectronSF");
                 const auto& muLepWeight  = tr.getVar<double>("totGoodMuonSF");
                 leptonScaleFactor = (GoodLeptons[0].first == "e") ? eleLepWeight : muLepWeight;
             }
+
+            const auto& bTagScaleFactor   = tr.getVar<double>("bTagSF_EventWeightSimple_Central");
+            const auto& htDerivedScaleFactor = tr.getVar<double>("htDerivedweight");
+            const auto& topPtScaleFactor = tr.getVar<double>("topPtScaleFactor");
+            const auto& prefiringScaleFactor = tr.getVar<double>("prefiringScaleFactor");
+
+            eventweight = lumi*Weight*puWeight;
```

Comment:
can you not add the puWeight to the eventweight. We have always called eventweight just the lumi*Weight.

Reply:
I would define a variable called weight = eventweight*puWeight.

Reply:
How does "theweight" sound?

Reply:
Sounds good